### PR TITLE
Assignment2

### PR DESCRIPTION
**Pull request by TechGlue**
Initial PR. 
Pair Code Partner: @NicholasJohnsonDev Nicholas Johnson

### CODE REVIEW

**Comment by Ashrom0** [comment]
I will review this

File: Logger/FileLogger.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+using System;
+using System.IO;
+
+namespace Logger
+{
+    public class FileLogger : BaseLogger
+    {
+        private string? ClassName { get; set; }
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:8]: This should have been declared in `BaseLogger.cs` and inherited here.

File: Logger/FileLogger.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            FilePath = filePath;
+        }
+
+        public FileLogger(string? className, string? filePath, LogLevel status, string? message)
+        {
+            ClassName = className;
+            FilePath = filePath;
+            Status = status;
+            Message = message;
+        }
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:25]: You should only have one constructor here that simply takes the file path. The rest of these parameters should not be class level properties, but rather scoped to only the log method.

File: Logger/FileLogger.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+        public string ToText()
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:27]: If you're going to separate this out, you may consider making it private as you only want people calling the `Log` method from this class.

File: Logger/FileLogger.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            return $"{DateTime.Now:MM/dd/yyyy hh:mm:ss tt}\n{ClassName}\n{Status}\n{Message}\n";
+        }
+
+        public override void Log(LogLevel logLevel, string message)
+        {
+            Status = logLevel;
+            Message = message;
+            string filePath = "";
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:36]: Why not just use `FilePath`? There isn't any need to create another variable for it here.

File: Logger/FileLogger.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            if (FilePath != null)
+            {
+                filePath = FilePath;
+            }
+            var fs = new FileStream(filePath, FileMode.Append, FileAccess.Write);
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:41]: Consider using a `using` here for your `FileStream` to automatically clean it up at the end of it's lifetime. Here is some documentation on it: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

File: Logger/FileLogger.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            StreamWriter sw = new(fs);
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:42]: Same thing here with the StreamWriter, `using` would be good. Also, you could just use the `StreamWriter`, no need for both.

File: Logger/FileLogger.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            sw.WriteLine(ToText());
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:43]: Forgetting to close the FileStream?

File: Logger/LogFactory.cs [hunk 8]
```diff
@@ -1,11 +1,16 @@
...
-            
-            return null;
-        }
-    }
-}
+﻿namespace Logger
+{
+    public class LogFactory
+    {
+        public string? FilePath { get; private set; }
```
Comment (mmwoodfo) [inline Logger/LogFactory.cs:5]: This looks good! 👍🏻 Nice private set.

File: Logger/BaseLoggerMixins.cs [hunk 9]
```diff
@@ -1,7 +1,55 @@
...
-    }
-}
+﻿using System;
+using System.Globalization;
+
+namespace Logger
+{
+    public static class BaseLoggerMixins
+    {
+        public static void Error(string? message, BaseLogger logger, params object[] arguments)
```
Comment (mmwoodfo) [inline Logger/BaseLoggerMixins.cs:8]:
```suggestion
        public static void Error(string? message, this BaseLogger logger, params object[] arguments)

```

File: Logger/BaseLoggerMixins.cs [hunk 10]
```diff
@@ -1,7 +1,55 @@
...
+        {
+            if (logger is null)
+                throw new ArgumentNullException(nameof(logger));
+
+            if (message != null)
```
Comment (mmwoodfo) [inline Logger/BaseLoggerMixins.cs:13]: You could also make it so `message` isn't nullable as a parameter.

File: Logger.Tests/LogFactoryTests.cs [hunk 11]
```diff
@@ -1,11 +1,42 @@
...
+    public class LogFactoryTests
+    {
+
+        [TestMethod]
+        public void ConfigureFileLogger_WithValidFilePath_PrivateMemberFilePathIsSet()
+        {
+            //Arrange
+            LogFactory logFactory = new();
+            //Act
+            logFactory.ConfigureFileLogger("/Users/luis/");
```
Comment (mmwoodfo) [inline Logger.Tests/LogFactoryTests.cs:13]: You may consider storing this string in a variable, so you don't have to type it twice.

**Comment by mmwoodfo** [comment]
@TechGlue Did you do a PR review for Assignment 2? I can't seem to find one.

**Pull request closed without merging**